File: src/my_project/medium_problems/from101to150/merge_intervals.py
from typing import List 

class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        out = []
        print(sorted(intervals, key=lambda i: i[0]))
        for i in sorted(intervals, key=lambda i: i[0]):
            if out and i[0] <= out[-1][-1]:
                out[-1][-1] = max(out[-1][-1], i[-1])
            else:
                out += i,
        return out
    
    
solution = Solution()

print(solution.merge([[2,6],[8,10],[1,3],[15,18]]))

File: my_project_typescript/easy_problems_typescript/from151to200/mergeIntervals.ts
function merge(intervals: number[][]): number[][] {

    let answer: number[][] = []

    intervals = intervals.sort((a,b) => a[0]<=b[0]?-1:1)

    for (let i of intervals){
        
        if (answer.length > 0 && (i[0] <= answer[answer.length-1][1])){
            answer[answer.length-1][1] = Math.max(answer[answer.length-1][1],i[1])
        }
        else{
            answer = answer.concat(Array(i))
        }
    }

    return answer

}